hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/util/Util.java

```diff
@@ -548,4 +550,51 @@ public static String formatJobDuration(long durationMs) {
         String textUpToHours = String.format("%02d:%02d:%02d.%03d", hours, minutes, seconds, millis);
         return sign + (durationMs > 0 ? durationMs + "d " : "") + textUpToHours;
     }
+
+    /**
+     * Given a list of input field names and a list of output field names
+     * creates a projection to map between these.
+     * <p>
+     * For example, if input names are {@code [surname, name, address]} and
+     * output names are {@code [name, surname, age]}, then the function,
+     * applied to {@code [Smith, John, New York]} will return {@code [John,
+     * Smith, (null)]}. That is, it will map the fields from the input order to
+     * output order. The output field named {@code age} is missing in input, so
+     * the value for it is {@code null} for any input.
+     *
+     * @param inputFields the input headers
+     * @param outputFields the output headers
+     * @return the indices to map input to output
+     */
+    @Nonnull
+    public static Function<String[], String[]> createFieldProjection(
+            @Nonnull String[] inputFields,
+            @Nonnull List<String> outputFields
+    ) {
+        if (outputFields.equals(asList(inputFields))) {
+            // shortcut - the mapping is an identity
+            return i -> i;
+        }
+        int[] simpleFieldMap = new int[outputFields.size()];
+        Arrays.fill(simpleFieldMap, -1);
+        for (int i = 0; i < inputFields.length; i++) {
+            int index = outputFields.indexOf(inputFields[i]);
+            // if the inputFields is present in the file and we didn't encounter it yet, store its index
+            if (index >= 0 && simpleFieldMap[index] == -1) {
```

Comment:
The false condition here isn't covered by tests.

Reply:
Both conditions are covered in integration tests. I added unit test coverage, if that's what you mean.